lib/src/Base/Optim/MultiStart.cxx

```diff
@@ -0,0 +1,160 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief MultiStart optimization algorithm
+ *
+ *  Copyright 2005-2016 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/MultiStart.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Cobyla.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/Box.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(MultiStart);
+
+static const Factory<MultiStart> Factory_MultiStart;
+
+/* Constructor with parameters */
+MultiStart::MultiStart()
+  : OptimizationSolverImplementation()
+  , solver_(new Cobyla)
+{
+  // Nothing to do here
+}
+
+
+/* Constructor with parameters */
+MultiStart::MultiStart(const OptimizationSolver & solver,
+                       const NumericalSample & startingPoints)
+  : OptimizationSolverImplementation(solver.getProblem())
+  , solver_(solver)
+{
+  setStartingPoints(startingPoints);
+}
+
+
+/* Check whether this problem can be solved by this solver.  Must be overloaded by the actual optimisation algorithm */
+void MultiStart::checkProblem(const OptimizationProblem & problem) const
+{
+  // The problem is checked in the local solver
+}
+
+
+void MultiStart::run()
+{
+  // run the solver with each starting point
+  OptimizationSolver solver(solver_);
+  resultCollection_.clear();
```

Comment:
Multistart::setProblem is not implemented and focus on :
 
void OptimizationSolverImplementation::setProblem(const OptimizationProblem & problem)
{
  checkProblem(problem);
  problem_ = problem;
}

Thus ever if we set a new problem it is not checked (Multistat::checkProblem is empty) and problem_ is never set explicitly to the local solver_